plugins/zenoh-plugin-storage-manager/src/replica/storage.rs

```diff
@@ -290,23 +298,31 @@ impl StorageService {
                 );
                 // there might be the case that the actual update was outdated due to a wild card update, but not stored yet in the storage.
                 // get the relevant wild card entry and use that value and timestamp to update the storage
-                let sample_to_store = match self
+                let sample_to_store: Sample = match self
                     .ovderriding_wild_update(&k, sample.timestamp().unwrap())
                     .await
                 {
-                    Some(overriding_update) => {
+                    Some(Update {
+                        kind: SampleKind::Put,
+                        data,
+                    }) => {
                         let Value {
                             payload, encoding, ..
-                        } = overriding_update.data.value;
-                        Sample::new(KeyExpr::from(k.clone()), payload)
-                            .with_encoding(encoding)
-                            .with_timestamp(overriding_update.data.timestamp)
-                            .with_kind(overriding_update.kind)
+                        } = data.value;
+                        PutSampleBuilder::new(KeyExpr::from(k.clone()), payload)
+                            .encoding(encoding)
+                            .timestamp(data.timestamp)
+                            .into()
                     }
-                    None => Sample::new(KeyExpr::from(k.clone()), sample.payload().clone())
-                        .with_encoding(sample.encoding().clone())
-                        .with_timestamp(*sample.timestamp().unwrap())
-                        .with_kind(sample.kind()),
+                    Some(Update {
+                        kind: SampleKind::Delete,
+                        data,
+                    }) => DeleteSampleBuilder::new(KeyExpr::from(k.clone()))
```

Comment:
From and API point of view, in my opinion would be better to have something like `SampleBuilder::delete(key) -> DeleteSampleBuilder`. But the user never explicitly uses a `DeleteSampleBuilder`. Opinions?

Reply:
done